### PR TITLE
Messages REST API

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [15490](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_backlogs/backlog/Life%20Events/Stories?workitem=15490)

### Description

Added a rest api for all basic message crud operations. (messages, templates and (email) providers).
Implemented in the SDK package
Replace the stubapi with sdk in web/messaging next apps.

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging/.env.sample [hunk 1]
```diff
@@ -8,4 +8,5 @@ POSTGRES_DB_NAME_SHARED=shared
 HOST_URL=http://localhost:3002/
 PAYMENTS_URL=http://localhost:3001/
 LOGIN_URL=http://localhost:8000/static/login/
-SENDGRID_API_KEY=
\ No newline at end of file
+SENDGRID_API_KEY=
+MESSAGES_BACKEND_URL=http://localhost:8002
```
Comment (Thurfjell) [inline apps/messaging/.env.sample:12]: @williamwd

File: apps/messaging-api/.env.local
Comment (Thurfjell) [inline apps/messaging-api/.env.local:1]: @williamwd

File: apps/messaging-api/openapitools.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (GiuliaTeggi) [inline apps/messaging-api/openapitools.json:1]: This file can be removed given we're moving away from the openapi-generator

File: apps/messaging-api/routes/messages/index.ts [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+        return { data };
+      } catch (err) {
+        // I guess we're gonna do some reporting with these errors?
+        const error: Static<typeof HttpError> = {
+          code: "no idea what this is supposed to be",
+          error: "failed to fetch messages",
+          message: "idk bro",
+          statusCode: 500,
+          time: new Date().toISOString(),
+        };
```
Comment (GiuliaTeggi) [inline apps/messaging-api/routes/messages/index.ts:134]: Don't think this is necessary right now, as we are going to have a common way to build and log errors, @SamSalvatico is working on it.

File: apps/messaging-api/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+COPY ./apps/messaging-api/ /app/apps/messaging-api/
+
+FROM deps AS builder
+WORKDIR /app
+
+ENV NODE_ENV=production
+ENV LOG_LEVEL=trace
+
+EXPOSE 8001
```
Comment (williamwd) [inline apps/messaging-api/Dockerfile]: Can you change to 8002, please?

Comment (williamwd) [reply apps/messaging-api/Dockerfile]: My bad, I see it's outdated now

**Review by williamwd — APPROVED** [review]

**Review by GiuliaTeggi — APPROVED** [review]
LGTM 🎉 🦜

**Pull request merged**